// n8n-nodes-qdrant/dist/nodes/Qdrant/descriptions/RetrievePoints.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.retrievePointsFields = exports.retrievePointsOperation = void 0;
const Commons_1 = require("./Commons");
exports.retrievePointsOperation = {
    name: 'Retrieve Points',
    value: 'retrievePoints',
    action: 'Retrieve Points',
    description: 'Retrieve all details from multiple points',
    routing: {
        request: {
            method: 'POST',
            url: '=/collections/{{$parameter.collectionName}}/points',
            body: {
                ids: '={{JSON.parse($parameter.ids)}}',
                with_payload: '={{$parameter.withPayload}}',
                with_vector: '={{$parameter.withVector}}',
                shard_key: '={{JSON.parse($parameter.shardKey)}}',
            },
            qs: {
                consistency: '={{$parameter.consistency}}',
                timeout: '={{$parameter.timeout}}',
            },
        },
    },
};
exports.retrievePointsFields = [
    (0, Commons_1.collectionNameField)('retrievePoints'),
    {
        displayName: 'IDs',
        name: 'ids',
        description: 'List of point IDs to retrieve',
        default: '[]',
        type: 'json',
        required: true,
        displayOptions: {
            show: {
                operation: ['retrievePoints'],
            },
        },
    },
    (0, Commons_1.withPayloadField)('retrievePoints'),
    (0, Commons_1.withVectorField)('retrievePoints'),
    (0, Commons_1.shardKeyField)('retrievePoints'),
    (0, Commons_1.consistencyField)('retrievePoints'),
    (0, Commons_1.timeoutField)('retrievePoints'),
];
